pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -935,3 +945,82 @@ func storePipelineSpec(ctx context.Context, pr *v1beta1.PipelineRun, ps *v1beta1
 	}
 	return nil
 }
+
+func (c *Reconciler) updatePipelineRunStatusFromInformer(pr *v1beta1.PipelineRun) error {
+	pipelineRunLabels := getTaskrunLabels(pr, "")
+	taskRuns, err := c.taskRunLister.TaskRuns(pr.Namespace).List(labels.SelectorFromSet(pipelineRunLabels))
```

Comment:
@afrittoli the `pipelineTask` associated to a condition (could be more than one condition as well) is not added to the `PipelineRunStatus` as its not executed/created until condition successfully finishes.

but this query returns `conditionCheckContainer` as well and it forces to iterate over those condition containers at line 961